packages/jsii-pacmak/test/generated-code/examples/nested-types.ts

```diff
@@ -0,0 +1,31 @@
+// See: https://github.com/aws/jsii/issues/2649
+
+export class Namespace1 {
+  public foo() {
+    return;
+  }
+}
+
+export class Namespace2 {
+  public foo() {
+    return;
+  }
+}
+
+export namespace Namespace1 {
+  export interface Foo {
+    readonly bar: string;
+  }
+
+  export interface IBar {
+    readonly bar: string;
+    method(): void;
+  }
+}
+
+export namespace Namespace2 {
+  export enum Foo {
+    BAR,
+    BAZ,
+  }
```

Comment:
Add a case of more than one nesting level (your code accounts for this)

Reply:
Done!